Core/Src/command_handler.c

```diff
@@ -266,6 +285,92 @@ void get_rtc_time(Iris_Timestamp *timestamp) {
 #endif
 }
 
+int transfer_image_to_nand(uint8_t sensor, uint8_t *file_timestamp) {
+    int ret = 0;
+    HAL_Delay(100);
+
+    uint32_t image_size;
+    image_size = read_fifo_length(sensor);
+
+    NAND_FILE *file = NANDfs_create();
+    if (!file) {
+        DBG_PUT("not able to create file %d failed: %d\r\n", file, nand_errno);
+        return -1;
+    }
+
+    int size_remaining;
+    uint8_t image[PAGE_DATA_SIZE];
+    uint32_t i = 0;
+    int chunks_to_write = ((image_size + (PAGE_DATA_SIZE - 1)) / PAGE_DATA_SIZE);
+
+    spi_init_burst(sensor);
+    for (int j = 0; j < chunks_to_write; j++) {
+        DBG_PUT("Writing chunk %d / %d\r\n", j + 1, chunks_to_write);
+        for (i = 0; i < PAGE_DATA_SIZE; i++) {
+            image[i] = spi_read_burst(sensor);
+        }
+        size_remaining = i;
+        while (size_remaining > 0) {
+            int size_to_write = size_remaining > PAGE_DATA_SIZE ? PAGE_DATA_SIZE : size_remaining;
+            ret = NANDfs_write(file, size_to_write, image);
+            if (ret < 0) {
+                DBG_PUT("not able to write to file %d failed: %d\r\n", file, nand_errno);
+                return -1;
+            }
+            size_remaining -= size_to_write;
+        }
+    }
+    spi_deinit_burst(sensor);
+    file->node.file_name = file_timestamp;
+
+    image_file_infos_queue[image_count].file_id = file->node.id;
+    image_file_infos_queue[image_count].file_name = file->node.file_name;
+    image_file_infos_queue[image_count].file_size = file->node.file_size;
+
+    ret = NANDfs_close(file);
+    if (ret < 0) {
+        DBG_PUT("not able to close file %d failed: %d\r\n", file, nand_errno);
+        return -1;
+    }
+
+    DBG_PUT("Image size: %d bytes\r\n", image_file_infos_queue[image_count].file_size);
+
+    DBG_PUT("%d|%s|%d", image_file_infos_queue[image_count].file_id, image_file_infos_queue[image_count].file_name,
+            image_file_infos_queue[image_count].file_size);
+
+    image_count += 1;
+    return 0;
+}
+
+int delete_image_file_from_queue(uint16_t index) {
+    image_file_infos_queue[index].file_id = 0xFF;
```

Comment:
you set file ID to 0xFF and then delete ID 0xFF?

Reply:
Yeah, that was a mistake. Should be fixed now